### PR TITLE
Vamshi/brain refactor

### PR DESCRIPTION
**Pull request by fazerlicourice7**
Brain refactor;
constants -> `config.h`
common code -> `common_fw`
specific code -> `___.cpp`

issue with pio though:
`common_fw.h` should go in `include/` and `common_fw.cpp` should go in `src`, but it fails to find all the relevant files; and stuff in common_fw doesn't exist. 
But if I put both in a lib folder, it works fine.

Doesn't work:
Screenshot from 2021-01-17 10-41-51

Works:
Screenshot from 2021-01-17 10-42-30

I don't care all that much about this diff, so I'm fine to merge in as a lib, but if you care, feel free to figure that out. 

This should be tested before merging.

### CODE REVIEW

**Comment by andycate** [comment]
If you want to import files from the `lib` directory, you use `<>`. If you want to import files from `include` then you use `""`.

**Comment by fazerlicourice7** [comment]
@andycate that doesn't work. (I'm assuming the .cpp file corresponding to the .h in include goes in src right?) It doesn't work.

**Comment by fazerlicourice7** [comment]
This has gotten pretty fucking fat... let's hope there aren't any errors, or it will be a lot of fun to test... I'd recommend doing some initial checks before the weekend if possible for ^^ that reason.

File: src/E1_coldflow/E1_coldflow.cpp [hunk 1]
```diff
@@ -12,7 +12,13 @@
...
 #include <batteryMonitor.h>
 
-#define RFSerial Serial6
+#define SERIAL_INPUT 1
+
+#if SERIAL_INPUT
+  #define RFSerial Serial
+#else
+  #define RFSerial Serial6
+#endif
```
Comment (fazerlicourice7) [inline src/E1_coldflow/E1_coldflow.cpp:21]: Was radio not working today?

**Review by taitmate38 — APPROVED** [review]
tested & doesn't blow up everything.

**Pull request merged**